### ISSUE TITLE
Remove (-/-) when no data entered for birth/death dates in author table

### ISSUE THREAD

**Issue by twinkietoes-on** [LibriVox/librivox-catalog#55]
In catalog lists and book pages

Examples: 
https://librivox.org/campfire-tales-of-jackson-hole-by-g-bryan-harry/
https://librivox.org/author/15350 (it's not showing under the author name, but still showing on the book listing)

Priority: 3

**Comment by RayBB** [comment]
[This](https://github.com/LibriVox/librivox-catalog/blob/58e96e31873435145c392a60eb05b964190c7114/application/helpers/previewer_helper.php#L198) seems to be the function in question.
It is only referenced three places in the codebase:
Author page - [under author's name](https://github.com/LibriVox/librivox-catalog/blob/58e96e31873435145c392a60eb05b964190c7114/application/views/catalog/author.php#L13)
Author page - [book listings](https://github.com/LibriVox/librivox-catalog/blob/58e96e31873435145c392a60eb05b964190c7114/application/views/catalog/item_blades/author_blade.php#L8)
Book page - [next to author's name](https://github.com/LibriVox/librivox-catalog/blob/58e96e31873435145c392a60eb05b964190c7114/application/views/catalog/item_blades/title_blade.php#L33)

It's not really clear to me how what might be causing the issue here since the usage is pretty similar.
The main difference seems to be that the two that are having issues are both [blades](https://laravel-guide.readthedocs.io/en/latest/blade/).

I'm not familiar enough with php or laravel to understand what is causing the difference when using blades.
My best guess is that the blades are being passed different data somehow.

**Comment by kgroeneveld** [comment]
I had started to look at this issue as well. I found at least four different functions that format the birth/death dates (and that isn't even including the one you linked to). I was planning on trying to unify them all as much as possible and have various changes sitting locally on my computer. Not sure when I will have time to finish it. I am easily distracting working on multiple things at once. I may look at it again this weekend.

**Comment by kgroeneveld** [comment]
I am really starting to go down the rabbit whole on this one...  

I have found 12 different spots that format the author dates now with many different variations on how it is implemented.

There are 4 examples in the JS code.  Two of these seemed to do the correct thing, one did not (it would show `(null - null)`) and one did not include the years at all when it should have.  I have consolidated all the JS uses down to one function.  While working on that I also found a couple bugs with author/translator suggestions on the new project page which got me a bit side tracked.

Now to try to figure out the 8 PHP examples and how they all fit together and hopefully consolidate them down to as few functions as possible...
